### PR TITLE
Removing star exports from @fluentui/react-link

### PR DESCRIPTION
**Pull request by khmakoto**
## Current Behavior

`react-link` has `export * from ...` in `src/index.ts`.

## New Behavior

`react-link` has explicitly named exports in `src/index.ts`.

## Related Issue(s)

#22099

### CODE REVIEW

**Comment by fabricteam** [comment]
## 📊 Bundle size report

<details>
<summary>Unchanged fixtures</summary>

| Package & Exports                                                                                                                                                             |         Size (minified/GZIP) |
| ----------------------------------------------------------------------------------------------------------------------------------------------------------------------------- | ---------------------------: |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/MultipleComponents.fixture.js'>react-components: Accordion, Button, FluentProvider, Image, Menu, Popover</abbr> | `175.69 kB`<br />`48.974 kB` |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/ProviderAndTheme.fixture.js'>react-components: FluentProvider & webLightTheme</abbr>                            | `32.526 kB`<br />`10.645 kB` |
| <samp>react-link</samp> <br /> <abbr title='bundle-size/Label.fixture.js'>Link</abbr>                                                                                         |  `11.102 kB`<br />`4.504 kB` |

</details>
<sub>🤖 This report was generated against <a href='https://github.com/microsoft/fluentui/commit/80746a7f3462c2fa4ff15a92d6e196c67d652f6a'>80746a7f3462c2fa4ff15a92d6e196c67d652f6a</a></sub>
<div id="bundleSizeComment9423"/>

**Comment by fabricteam** [comment]
## [Perf Analysis (`@fluentui/react-components`)](https://github.com/microsoft/fluentui/wiki/Perf-Testing)

  <table>
  <tr>
    <th>Scenario</th>
    <th>Render type</th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">Master Ticks</a>
    </th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">PR Ticks</a>
    </th>
    <th>Iterations</th>
    <th>Status</th>
  </tr><tr>
            <td>FluentProviderWithTheme</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22112/perf-test-react-components/FluentProviderWithTheme-mount_base.html">239</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22112/perf-test-react-components/FluentProviderWithTheme-mount.html">307</a></td>
            <td>10</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22112/perf-test-react-components/FluentProviderWithTheme-mount.regression.txt">Possible regression</a></td>
           </tr></table><details><summary>All results</summary><p>
  <table>
  <tr>
    <th>Scenario</th>
    <th>Render type</th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">Master Ticks</a>
    </th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">PR Ticks</a>
    </th>
    <th>Iterations</th>
    <th>Status</th>
  </tr><tr>
            <td>Avatar</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22112/perf-test-react-components/Avatar-mount_base.html">830</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22112/perf-test-react-components/Avatar-mount.html">799</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Button</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22112/perf-test-react-components/Button-mount_base.html">477</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22112/perf-test-react-components/Button-mount.html">480</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProvider</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22112/perf-test-react-components/FluentProvider-mount_base.html">1674</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22112/perf-test-react-components/FluentProvider-mount.html">1729</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22112/perf-test-react-components/FluentProviderWithTheme-mount_base.html">239</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22112/perf-test-react-components/FluentProviderWithTheme-mount.html">307</a></td>
            <td>10</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22112/perf-test-react-components/FluentProviderWithTheme-mount.regression.txt">Possible regression</a></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22112/perf-test-react-components/FluentProviderWithTheme-virtual-rerender_base.html">174</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22112/perf-test-react-components/FluentProviderWithTheme-virtual-rerender.html">214</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender-with-unmount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22112/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount_base.html">311</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22112/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount.html">273</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>MakeStyles</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22112/perf-test-react-components/MakeStyles-mount_base.html">1345</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22112/perf-test-react-components/MakeStyles-mount.html">1382</a></td>
            <td>50000</td>
            <td></td>
           </tr></table></p></details>

<div id="perfComment9425"/>

**Review by ecraig12345 — APPROVED** [review]

File: packages/react-link/src/index.ts [hunk 1]
```diff
@@ -1 +1,9 @@
-export * from './Link';
+export {
+  Link,
+  linkClassName,
```
Comment (spmonahan) [inline packages/react-link/src/index.ts:3]: There is also [linkClassNames](https://github.com/microsoft/fluentui/blob/master/packages/react-link/src/components/Link/useLinkStyles.ts#L11) now.

Comment (khmakoto) [reply packages/react-link/src/index.ts:3]: Added.

**Pull request merged**